### PR TITLE
Fix/login

### PR DESCRIPTION
**Pull request by fedgut**
Quick Info
Adapt HR's TIL to make it a bit more modern and adapt the look and feel to MagmaLabs

Migrations?  Yes

What does this change?
-  Updates Ruby to 2.4.6
-  General updates to gems 
-  Change Autehm + Omniauth authentication for Devise
-  Adapt styles 
-  Add Rubocop

Why are you changing that?
The previous log in and sessions system was no longer functional
- The Omniauth gem was outdated and would not work with google's API. 
- The Authem gem's documentation is no longer available and the gem is not maintained.   

Changing Authem, for Devise was the best approach to make the TIL functional

How did you accomplish this change?
- Fix compatibility and deprecation issues from updating gems
- Remove Authem and Omniauth from controller, models, and migrations
- Add Devise to the developer model
- Fix styles

Merge/Deploy Checklist
    It has been reviewed and accepted.

How do you manually test this?
     Create an account, create a post and publish it or save it as a draft.

Screenshots

Desktop 
Yes

Mobile 
Not yet fully tested, just some style fixes and verifications.

### CODE REVIEW

File: app/views/posts/edit.html.haml [hunk 1]
```diff
@@ -3,10 +3,12 @@
...
   = render 'form'
 
-  %article.post
-    %section
-      .post__content.copy
-        %h1.title_preview
-        .content_preview
-        %footer
-          %p.word_count
+-# Post preview can be enabled by uncommenting below
```
Comment (mumoc) [inline app/views/posts/edit.html.haml:6]: if this is not used anymore, remove the comment

**Review by mumoc — CHANGES_REQUESTED** [review]
In general, LGTM. Check the color value I commented; and remove commented code, we don't need it. 

Something that you can do is to convert all the erb files to haml using the rake task at least: `rake haml:erb2haml`

File: app/assets/stylesheets/_variables.scss [hunk 2]
```diff
@@ -12,13 +13,14 @@ $cyan: #33d6ea;
...
 $blue: #67acdb;
+$bg-red: #d92f42;
 
 $twitter-color: #55acee;
 $github-color: #4b4b4b;
 
 $base: $dark;
 $bg: $light;
-$accent: $navy;
+$accent: #ffffffef;
```
[the remark this answers is not in the document]
Comment (fedgut) [reply app/assets/stylesheets/_variables.scss:23]: the additional chars are for transparency!

File: app/views/posts/edit.html.haml (hunk 1 above)
Comment (fedgut) [reply app/views/posts/edit.html.haml:6]: gotcha!

**Pull request merged**